### DISCUSSION TITLE
Add an object serialization API

### DISCUSSION THREAD

**Discussion by KeitoTobi1** [vlang/v#11624] [Ideas]
Add an object serialization API similar to Go's Gob and Python's Pickle

**Comment by alula** [comment]
dunno what you meant, V only supports serializing to JSON but it's very limited (lacks type information and etc.)

**Comment by medvednikov** [comment]
I think that's what they are saying, they'd like V to have its own serialization API, like Go's Gob.

@keito940 it'll definitely be implemented next year :)

**Comment by dumblob** [comment]
Wouldn't it be easier to stick with an existing in-memory format like [sproto](https://github.com/cloudwu/sproto ) or [Cap'n Proto](https://capnproto.org/ ) instead of reinventing the wheel (like Go Gob kind of does)?

**Comment by medvednikov** [comment]
Yes, maybe.
